docs/src/main/asciidoc/resteasy-reactive-virtual-threads.adoc

```diff
@@ -0,0 +1,308 @@
+
+    static final double VALENCE_LATITUDE = 44.9;
+    static final double VALENCE_LONGITUDE = 4.9;
+
+    static final double ATHENS_LATITUDE = 37.9;
+    static final double ATHENS_LONGITUDE = 23.7;
+
+    @RestClient WeatherService service;
+
+    @GET
+    @RunOnVirtualThread <1>
+    public String getTheBestPlaceToBe() {
+        var valence = service.getWeather(VALENCE_LATITUDE, VALENCE_LONGITUDE).weather().temperature();
+        var athens = service.getWeather(ATHENS_LATITUDE, ATHENS_LONGITUDE).weather().temperature();
+
+        // Advanced decision tree
+        if (valence > athens && valence <= 35) {
+            return "Valence! (" + Thread.currentThread() + ")";
+        } else if (athens > 35) {
+            return "Valence! (" + Thread.currentThread() + ")";
+        } else {
+            return "Athens (" + Thread.currentThread() + ")";
+        }
+    }
+}
+----
+<1> Instructs Quarkus to invoke this method on a virtual thread
+
+== Run the application in dev mode
+
+Make sure you use JDK and JVM supporting virtual thread and launch the _dev mode_ with `./mvnw quarkus:dev`:
+
+[source, shell]
+----
+> java --version
+openjdk 20.0.1 2023-04-18 <1>
+OpenJDK Runtime Environment Temurin-20.0.1+9 (build 20.0.1+9)
+OpenJDK 64-Bit Server VM Temurin-20.0.1+9 (build 20.0.1+9, mixed mode)
+
+> ./mvnw quarkus:dev <2>
+----
+<1> Must be 19+
+<2> Launch the dev mode
+
+Then, in a browser, open http://localhost:8080.
+You should get something like:
+
+[source, text]
+----
+Valence! (VirtualThread[#144]/runnable@ForkJoinPool-1-worker-6)
+----
+
+As you can see, the endpoint runs on a virtual thread.
+
+It's essential to understand what happened behind the scene:
+
+1. Quarkus creates the virtual thread to invoke your endpoint (because of the `@RunOnVirtualThread` annotation).
+2. When the code invokes the rest client, the virtual thread is blocked, BUT the carrier thread is not blocked (that's the virtual thread _magic touch_).
+3. Once the first invocation of the rest client completes, the virtual thread is rescheduled and continues its execution.
+4. The second rest client invocation happens, and the virtual thread is blocked again (but not the carrier thread).
+5. Finally, when the second invocation of the rest client completes, the virtual thread is rescheduled and continues its execution.
+6. The method returns the result. The virtual thread terminates.
+7. The result is captured by Quarkus and written in the HTTP response
+
+== Verify pinning using tests
+
+In the `pom.xml,` we added an `argLine` argument to the surefire and failsafe plugins:
+
+[source, xml]
+----
+<argLine>--enable-preview -Djdk.tracePinnedThreads</argLine>
+----
+
+The `-Djdk.tracePinnedThreads` dumps the stack trace if a virtual thread cannot be _unmounted_ smoothly (meaning that it blocks the carrier thread).
+That's what we call _pinning_ (more info in xref:./virtual-threads#pinning[the virtual thread reference guide]).
+
+We recommend enabling this flag in tests.
```

Comment:
It would be cool if we had some integration with our `@QuarkusTest` that would allow users to retrieve them and assert on them.  Obviously it's unrelated to this PR

Reply:
:)